CHANGELOG.md

```diff
@@ -6,9 +6,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ## StreamChat
 ### ⚡ Performance
 - Use background database observers for `CurrentUserController.currentUser`, `ChatChannelMemberListController.members`, and `ChatMessageController.message` which reduces CPU usage on the main thread [#3357](https://github.com/GetStream/stream-chat-swift/pull/3357)
+- Reduce latency of the `BackgroundDatabaseObserver` by reducing thread switching when handling changes [#3373](https://github.com/GetStream/stream-chat-swift/pull/3373)
```

Comment:
@laevandus There is a -120.24% decrease in performance of the Channel List, we should re-run the benchmark to make sure it was not false positive

Reply:
I'll rerun it but locally when profiling with Instruments I got 0 hangs on my iPhone 12 Pro Max. No DB reads on the main thread either in `BackgroundDatabaseObserver`.

Reply:
Ran it manually [here](https://github.com/GetStream/stream-chat-swift/actions/runs/10384123700/job/28750433059)

Reply:
One red, others green. Seems like the perf test is a bit random.